src/deprecated.jl

```diff
@@ -222,3 +222,8 @@ import Base: names
 ### Changed in PR#694
 @deprecate has(parent::Union{File,Group,Dataset}, path::AbstractString) Base.haskey(parent, path)
 @deprecate exists(parent::Union{File,Group,Dataset,Datatype,Attributes}, path::AbstractString) Base.haskey(parent, path)
+
+function Base.read(f::Base.Callable, parent::DataFile, name::AbstractString...)
+    depwarn("Base.read(f::Base.Callable, parent::H5DataType, name::AbstractString...) is deprecated. Directly call `f` on the output from `read(parent, name...)`", :read)
+    f(read(parent, name...)...)
+end
```

Comment:
I just noticed that there's a similar `g_create` method that should probably also be deprecated: https://github.com/JuliaIO/HDF5.jl/blob/1097227fcb66c65f38b916af11645c104316f6e0/src/HDF5.jl#L717-L724

Reply:
Good catch.

Seems like this is supposed to be used as follows

```
h5rewrite(outfile) do fid
    g_create(fid, "mygroup") do g
        write(g, "x", 3.3)
    end
end
```
which kinda strikes me as odd